### ISSUE TITLE
onClose was called on null

### ISSUE THREAD

**Issue by philipgiuliani** [mfeckie/phoenix_wings#8]
It probably happened when calling `socket.disconnect()` before its full connected. But I am not sure about that.

```
E/flutter ( 4834): [ERROR:flutter/shell/common/shell.cc(181)] Dart Error: Unhandled exception:
E/flutter ( 4834): NoSuchMethodError: The method 'onClose' was called on null.
E/flutter ( 4834): Receiver: null
E/flutter ( 4834): Tried calling: onClose(Closure: () => dynamic from Function 'reconnect':.)
E/flutter ( 4834): #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)
E/flutter ( 4834): #1      PhoenixSocket.connect (package:phoenix_wings/src/phoenix_socket.dart:107:11)
E/flutter ( 4834): #2      _RootZone.runUnary (dart:async/zone.dart:1379:54)
E/flutter ( 4834): #3      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
E/flutter ( 4834): #4      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
E/flutter ( 4834): #5      Future._propagateToListeners (dart:async/future_impl.dart:671:32)
E/flutter ( 4834): #6      Future._complete (dart:async/future_impl.dart:476:7)
E/flutter ( 4834): #7      _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
E/flutter ( 4834): #8      _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
E/flutter ( 4834): #9      _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:295:13)
E/flutter ( 4834): #10     PhoenixIoConnection.waitForConnection (package:phoenix_wings/src/phoenix_io_connection.dart)
E/flutter ( 4834): #11     _RootZone.runUnary (dart:async/zone.dart:1379:54)
E/flutter ( 4834): #12     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
E/flutter ( 4834): #13     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
E/flutter ( 4834): #14     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
E/flutter ( 4834): #15     Future._addListener.<anonymous closure> (dart:async/future_impl.dart:351:9)
E/flutter ( 4834): #16     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
E/flutter ( 4834): #17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
```

**Comment by mfeckie** [comment]
Is this still an issue?

**Comment by mfeckie** [comment]
Should be fixed in 0.2.1
